yellowbrick/target/feature_correlation.py

```diff
@@ -0,0 +1,87 @@
+# yellowbrick.classifier.feature_correlation
+# Class balance visualizer for showing per-class support.
+#
+# Author    Zijie (ZJ) Poh <poh.zijie@gmail.com>
+# Author:   Benjamin Bengfort <bbengfort@districtdatalabs.com>
+# Created:  Wed Jul 29 15:30:40 2018 -0700
+#
+# ID: feature_correlation.py [5388065] neal@nhumphrey.com $
+
+"""
+Feature Correlation to Dependent Variable Visualizer.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+import numpy as np
+
+from .base import TargetVisualizer
+
+from sklearn.feature_selection import (mutual_info_classif,
+                                       mutual_info_regression)
+from scipy.stats import pearsonr
+
+
+##########################################################################
+## Class Feature Correlation
+##########################################################################
+
+class FeatureCorrelation(TargetVisualizer):
+
+    method_func = {
+        'pearson': lambda X, y: [pearsonr(x, y)[0] for x in X.T],
+        'mutual_info': lambda X, y: mutual_info_regression(X, y),
+        'mutual_info_classif': lambda X, y: mutual_info_classif(X, y)
+    }
+    method_label = {
+        'pearson': 'Pearson Correlation',
+        'mutual_info': 'Mutual Information',
+        'mutual_info_classif': 'Mutual Information'
+    }
+
+    def __init__(self, ax=None, method='pearson', classification=False,
+                 **kwargs):
+        super(FeatureCorrelation, self).__init__(ax=None, **kwargs)
+
+        if classification and method == 'mutual_info':
+            method = 'mutual_info_classif'
```

Comment:
I'm glad you inserted this check and thought about how to specify the difference between the two mutual information algorithms. Another possibility is to use `target_type` on fit to determine if the output is `continuous` or `binary`, `multiclass` etc. and you can choose the metric that way. 

Also, I'd recommend that you do a check for `method in self.method_func` and if not, raise an exception `YellowbrickValueError` when the visualizer is instantiated.

Reply:
Hmm.... Can you explain how to determine if we are dealing with a classification or a regression problem from the target type? For example, y should be `int` in classification problem but having y as `float` will still work. Also, if `y` is age, then it can be `int` and also regression problem. So, perhaps is better for the user to specify the problem type, albeit more work and less elegant.

Reply:
You're right that in general the type of model that the user selects will determine how the model treats the target. In this case, however, the difference between [`mutual_info`](http://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.mutual_info_regression.html#sklearn.feature_selection.mutual_info_regression) and [`mutual_info_classif`](http://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.mutual_info_classif.html#sklearn.feature_selection.mutual_info_classif) is the solver that relies on a continuous or discrete input and will only provide interpretable results based on the form of the target. 

Let's look at all the cases:

- y is continuous: the discrete solver will not work, classification will also not work because of the huge number of possible classes (binning or thresholding is required) 
- y is discrete, small number of classes: the continuous solver will not work, also a regression model would probably not behave sensibly 
- y is discrete, large number of classes: both solvers will perform poorly, though the discrete solver would probably perform a bit better. Both regression and classification models are possible, perhaps regression models (e.g. for age) would be a bit better without some binning into age groups 

Since we don't have a model to check, and because scikit-learn does use [`target_type`](http://scikit-learn.org/dev/modules/generated/sklearn.utils.multiclass.type_of_target.html) to raise exceptions for badly formed problems, I would err on the side of the first two common cases rather than create a possible confusion to handle the third case.